web/documentserver-example/nodejs/public/javascripts/jscript.js

```
@@ -293,6 +334,97 @@ if (typeof jQuery != "undefined") {
+        });
+        jq("#hiddenFileName").val(fileName);
+        jq("#convertStep1").addClass("done");
+        jq("#convertStep2").addClass("waiting");
+    });
+
+    jq(document).on("click", "td[name='convertingTypeButton']:not(.disable, .orange)", function () {
+        const currentElement = jq(this);
+        let id = currentElement[0].id;
+        let fileExt = jq(`#${id}`).attr("data");
+        jq(`#${id}`).addClass("orange");
+        jq("td[name='convertingTypeButton']").addClass("disable");
+        jq("#convertStep2").removeClass("waiting").removeClass("done").addClass("current");
+        jq("#convertStep2").text('2. File conversion');
+        jq("#convert-descr").removeClass("disable");
+        jq("#convertPercent").text("0 %");
+        jq("#hiddenFileName").attr("placeholder",fileExt);
+        jq("#downloadConverted").addClass("disable");
+        jq("#beginEditConverted").addClass("disable");
+        jq("#beginViewConverted").addClass("disable");
+        mustReload = true;
+        
+        convertFile();
+    });
+
+    function convertFile (filePass) {
+        let fileName = decodeURIComponent(jq("#hiddenFileName").val());
+        let fileExt = jq("#hiddenFileName").attr("placeholder");
+
+        if (timer != null) {
+            clearTimeout(timer);
+        }
+        timer = setTimeout(function () {
+            jq.ajaxSetup({ cache: false });
+            jq.ajax({
+                async: true,
+                type: "post",
+                dataType: "json",
+                data: {filename: fileName, filePass: filePass, lang: language, fileExt: fileExt},
+                url: UrlConverter,
+                complete: function (data) {
+                    try {
+                        var response = jq.parseJSON(data.responseText);
+                    } catch (e)	{
+                        response = { error: e };
+                    }
```

The error is not handled in any way. When an error is returned, we simply write it to `response`, but do not process it further.
We need something like `if (response.error) {...}` At least call `return`